android/src/com/mapswithme/maps/location/SensorHelper.java

```diff
@@ -39,46 +37,54 @@ public void onSensorChanged(SensorEvent event)
     if (!MwmApplication.get().arePlatformAndCoreInitialized())
       return;
 
-    if (mCompassParams.isDataAbsent())
-      notifyCompassImmediately(event);
+    if (mSensorData.isAbsent())
+    {
+      notifyImmediately(event);
+      return;
+    }
+
+    SensorType type = SensorType.get(event);
+    if (type == SensorType.TYPE_MAGNETIC_FIELD)
+      addRateLimitMessage();
 
-    else if (!mHandler.hasMessages(MARKER))
-      notifyCompass(event);
+    if (!mHandler.hasMessages(MARKER) || type != SensorType.TYPE_MAGNETIC_FIELD)
```

Comment:
!mHandler.hasMessage(MARKER) will never be true as you call addRateLimitMessage(); at line 48.
So SensorType.TYPE_MAGENT_FIELD SensorEvents will never be handled by notifyImmediately(...).

Messages previously handled by SensorEventHandler handleMessage(Message) called notifyImmediatly,
but the implementation has now been removed by using Handler directly ....